modules/electrophysiology_browser/ajax/get_electrophysiology_session_data.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * Get EEG Session Data.
+ *
+ * This retrieves the entire eeg session data
+ * for a candidate to be visible on the eeg_session form.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  EEG_Session
+ * @author   Muhammad Khan <muhammad.khan@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris-Trunk
+ */
+
+require_once 'BIDSFile.class.inc';
+
+/**
+ * User permission verification:
+ * Only users that have the eeg view all sites permission OR
+ * have the view site permission and belong to candidates session site
+ * are able to fetch the eeg session data.
+ */
+$user =& \User::singleton();
+$timePoint =& \TimePoint::singleton($_REQUEST['sessionID']);
+if (!$user->hasPermission('electrophysiology_browser_view_allsites')
```

Comment:
This could probably be rewritten to be more clear. It should at least have a comment explaining the permissions.

Reply:
I added a comment explaining it. Hopefully that helps clarifying. Not sure how to clarify the code itself better...